frontends/concrete-python/concrete/fhe/mlir/context.py

```diff
@@ -2173,6 +2188,22 @@ def encrypt(self, resulting_type: ConversionType, x: Conversion) -> Conversion:
     def equal(self, resulting_type: ConversionType, x: Conversion, y: Conversion) -> Conversion:
         return self.comparison(resulting_type, x, y, accept={Comparison.EQUAL})
 
+    def shift_left(self, x: Conversion, rank):
+        assert rank >= 0
+        assert rank < x.bit_width
+        shifter = 2**rank
+        shifter = self.constant(self.i(x.bit_width + 1), shifter)
+        return self.mul(x.type, x, shifter)
+
+    def reduce_precision(self, x: Conversion, bit_width):
+        assert bit_width > 0
+        assert bit_width <= x.type.bit_width
+        if bit_width == x.type.bit_width:
```

Comment:
Use `x.bit_width` directly instead of `x.type.bit_width`.

Reply:
Same with other usages below.

Reply:
Done